### ISSUE TITLE
NonNull annotation needed in onMapReady MapboxMap parameter

### ISSUE THREAD

**Issue by cammace** [mapbox/mapbox-gl-native#4429]
Writing this: 

    mapView.getMapAsync(new OnMapReadyCallback() {
                @Override
                public void onMapReady(final MapboxMap mapboxMap) {

gives the warning 

> Not annotated parameter overrides @NonNull parameter

but adding the @NonNull annotation resolves the Android Studio warning. Can we get rid of the `@NonNull` annotation requirement?

@zugaldia @bleege @tobrun

**Comment by tobrun** [comment]
It's actually already there..
It should automatically add that..

```java
package com.mapbox.mapboxsdk.maps;

import android.support.annotation.NonNull;

/**
 * Interface definition for a callback to be invoked when the map is ready to be used.
 * <p>
 * Once an instance of this interface is set on a {@link MapFragment} or {@link MapView} object,
 * the onMapReady(MapboxMap) method is triggered when the map is ready to be used and provides a non-null instance of {@link MapboxMap}.
 * </p>
 */
public interface OnMapReadyCallback {

    /**
     * Called when the map is ready to be used.
     *
     * @param mapboxMap A non-null instance of a MapboxMap associated with the {@link MapFragment}
     *                  or {@link MapView} that defines the callback.
     */
    void onMapReady(@NonNull MapboxMap mapboxMap);
}


```

**Comment by tobrun** [comment]
I have no issue removing it

**Comment by tobrun** [comment]
Talked with @cammace about it, he is going to pick this up :rocket:

**Comment by bleege** [comment]
Rebased and merged into `release-ios-3.2.0-android-4.0.0`.
